mlflow/genai/evaluation/harness.py

```diff
@@ -124,6 +124,14 @@ def _run_single(
     assessments.extend(eval_item.get_expectation_assessments())
     eval_result = EvalResult(eval_item=eval_item, assessments=assessments)
 
+    tags = eval_item.tags or {}
```

Comment:
should we add a type validation guard here to ensure that users aren't passing items other than the expected structure with a helpful exception message about the required format for tags?